### DISCUSSION TITLE
Tracing through bridged interfaces

### DISCUSSION THREAD

**Discussion by ktims** [netbox-community/netbox#9076] [Q&A]
It looks like tracing through two-member bridges should be supported according to #7761 . However this doesn't seem to be working for me; I must be misunderstanding something. I have tried (on my 3.2 installation, as well as the demo instance):

* One way bridge directly between two interfaces (e.g. ge-0/0/0 bridged to ge-0/0/1). Tracing neither direction passes through the bridge.
* Two way bridge directly between two interfaces (e.g. ge-0/0/0 bridge to ge-0/0/1 and ge-0/0/1 bridged to ge-0/0/0). Same result.
* Virtual bridge interface with both interfaces bridged to it. Same result as well.

I did also run the `migrate.py trace_paths --force` in case there was some stale state from before this feature was added, but no change.

How is this meant to work? I some additional 802.1Q configuration required or something?

**Comment by jeremystretch** [comment]
It works as expected for me on v3.2.0:

1. Create 3 devices w/interfaces
2. On device 2, edit interface A so that it bridges to B, and set B to bridge to A
3. Connect device1 A to device2 A
4. Connect device2 B to device3 B
5. Trace from either endpoint

![Screenshot 2022-04-08 at 08-26-19 Cable Trace for Interface GigabitEthernet1 0 1 NetBox](https://user-images.githubusercontent.com/13487278/162435258-f6928c9f-38e9-42f3-a3bf-fc8ce6e4b01e.png)

**Reply by ktims** [reply]
Okay thanks, I think I see that here is some discrepancy based on where the trace is started from, possibly intentional though it's a bit unexpected. I created a test setup in the demo instance, adding a device nidA with ports 1/0/1 and 1/0/2 bridged and connected to one existing device and one circuit termination in the existing test dataset:

![image](https://user-images.githubusercontent.com/1471533/162516435-ba022c44-0f8c-4426-92d0-88ccef936fd6.png)

If I originate the trace from either nidA interface, the bridge is not included, which seems a somewhat reasonable behaviour, though a bit unexpected:

![image](https://user-images.githubusercontent.com/1471533/162516698-33dc8836-e2b5-4316-8696-8bfc89821692.png)

Originating the trace from the router port works as expected and matches the behaviour you've shown.

However originating the trace from the circuit termination doesn't flow through the bridge either:

![image](https://user-images.githubusercontent.com/1471533/162516979-accfb49e-079c-425e-b494-6597ccac4a08.png)

Where I originally noticed this there are extra cables and patch panels in the paths, but the behaviour is essentially the same, starting the trace from a Circuit or the interface on the device owning the bridge doesn't connect the bridged ports.

I also notice in poking around this that both in the Cable page and the Interfaces listing only the Provider or Provider Network is linked. It would be nice to have a link to the Circuit or Circuit Termination here as well, similar to how both the Device and Interface are linked in those terminations.
